src/radix-tree/radix.ts

```diff
@@ -0,0 +1,235 @@
+import { create as createNode, Node, updateParent, removeDocument } from "./node";
+import { boundedLevenshtein } from "../levenshtein";
+import { getOwnProperty } from "../utils";
+
+export type Nodes = Record<string, Node>;
+
+export type FindParams = {
+  term: string;
+  exact?: boolean;
+  tolerance?: number;
+};
+
+export type FindResult = Record<string, string[]>;
+
+export function insert(root: Node, word: string, docId: string) {
+  for (let i = 0; i < word.length; i++) {
+    const currentCharacter = word[i];
+    root.docs.push(docId);
+
+    if (currentCharacter in root.children) {
+      const edgeLabel = root.children[currentCharacter].word;
+      const commonPrefix = getCommonPrefix(edgeLabel, word.substring(i));
```

Comment:
Please extract all repeatedly computed values in variables:

* `root.children[currentCharacter]`
* `word.substring(i)`
* `word.substring(i).length`

and so forth